Oracle NetSuite Entities & Activities Sources documentation initial commit

mohan-h opened the pull request.

mohan-h committed:
Oracle NetSuite Activities Source initial commit

mohan-h committed:
Mask vendor information

mohan-h committed:
Oracle NetSuite Entities initial commit

mohan-h committed:
Refactored including Oracle NetSuite Entities

carlojavier commented on help/sources/connectors/marketing-automation/oracle-netsuite.md:
```diff
@@ -0,0 +1,73 @@
+* Have an active subscription to any product.
+* [!DNL Oracle NetSuite] uses OAuth 2.0 to communicate with [!DNL Oracle NetSuite] API's. 
+    * You will need your **Account ID**. Refer to the [Finding your NetSuite Account ID](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_1498754928.html#Finding-Your-NetSuite-Account-ID)] page if you do not know your NetSuite Account ID.
+    * You will need your **[!DNL NetSuite] Account ID**. Refer to the [Finding your NetSuite Account ID](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_1498754928.html#Finding-Your-NetSuite-Account-ID)] page if you do not know your NetSuite Account ID.
```
I think we just need one of these bullet points. We can just keep the first one.

carlojavier commented on help/sources/connectors/marketing-automation/oracle-netsuite.md:
```diff
@@ -0,0 +1,73 @@
+Before you can bring your [!DNL Oracle NetSuite] data to Experience Platform, you must first ensure that you have the following:
+
+* A [!DNL Oracle NetSuite] account. If you do not already have an account, contact [!DNL Oracle NetSuite](https://www.NetSuite.com/portal/company/contactus.shtml).
+* Have an active subscription to any product.
```
I think we can revise this to: "An active subscription to any [!DNL Oracle NetSuite] product." Or should it be "... any [!DNL Oracle] product."

carlojavier commented on help/sources/connectors/marketing-automation/oracle-netsuite.md:
```diff
@@ -0,0 +1,73 @@
+* [!DNL Oracle NetSuite] uses OAuth 2.0 to communicate with [!DNL Oracle NetSuite] API's. 
+    * You will need your **Account ID**. Refer to the [Finding your NetSuite Account ID](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_1498754928.html#Finding-Your-NetSuite-Account-ID)] page if you do not know your NetSuite Account ID.
+    * You will need your **[!DNL NetSuite] Account ID**. Refer to the [Finding your NetSuite Account ID](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_1498754928.html#Finding-Your-NetSuite-Account-ID)] page if you do not know your NetSuite Account ID.
+    * To access the [!DNL Oracle NetSuite] API's, You will need a **Client ID** and **Client secret**, follow the [OAuth 2.0](https://docs.oracle.com/en/cloud/saas/NetSuite/ns-online-help/chapter_157769826287.html#OAuth-2.0) tutorial within your [!DNL Oracle NetSuite] account and ensure you execute the below:
```
replace "... the below:" with "... the following:"

carlojavier commented on help/sources/connectors/marketing-automation/oracle-netsuite.md:
```diff
@@ -0,0 +1,73 @@
+    * To access the [!DNL Oracle NetSuite] API's, You will need a **Client ID** and **Client secret**, follow the [OAuth 2.0](https://docs.oracle.com/en/cloud/saas/NetSuite/ns-online-help/chapter_157769826287.html#OAuth-2.0) tutorial within your [!DNL Oracle NetSuite] account and ensure you execute the below:
+        * Ensure your [!DNL Oracle NetSuite] administrator has enabled the OAuth 2.0 feature
+        * Setup Roles
+        * Assigned Users to the Oauth 2.0 Roles, and created Integration records
```
We can revise some of these to sentence-case:
- Ensure your [!DNL Oracle NetSuite] administrator has enabled the OAuth 2.0 feature.
- Set up roles.
- Assign users to OAuth 2.0 roles and create integration records.

mohan-h replied:
Have merged 1 & 2

carlojavier commented on help/sources/connectors/marketing-automation/oracle-netsuite.md:
```diff
@@ -0,0 +1,73 @@
+        * Ensure your [!DNL Oracle NetSuite] administrator has enabled the OAuth 2.0 feature
+        * Setup Roles
+        * Assigned Users to the Oauth 2.0 Roles, and created Integration records
+    * You will also need to generate an **Access token** and a **Refresh Token**. Refer to the [OAuth 2.0 Authorization Code Grant Flow](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_158074210415.html#OAuth-2.0-Authorization-Code-Grant-Flow) tutorial to generate them. Complete [Step One](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_158081944642.htm ...
```
- we can refer to "access token" and "refresh token" in sentence case in this context
- sentence case for both "step one" and "step two"

carlojavier commented on help/sources/connectors/marketing-automation/oracle-netsuite.md:
```diff
@@ -0,0 +1,73 @@
+| authorizationTestUrl | Your [!DNL NetSuite] authorization test URL. This is optional. | <code>https://<NETSUITE_ACCOUNT_ID>.app.netsuite.com<br>/app/login/oauth2/authorize.nl?response_type=code<br>&redirect_uri=https%3A%2F%2Fapi.github.com<br>&scope=rest_webservices<br>&state=ykv2XLx1BpT5Q0F3MRPHb94j<br>&client_id=<CLIENT_ID></code> |
+| Access token | The Access token value is generated at the end of [Step Two](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_158081952044.html#Step-Two-POST-Request-to-the-Token-Endpoint) of the [OAuth 2.0 Authorization Code Grant Flow](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_158074210415.html#OAuth-2.0-Authorization-Code-Grant-Flow) tutorial.  ...
+| Refresh token | The Refresh token value value is generated at the end of [Step Two](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_158081952044.html#Step-Two-POST-Request-to-the-Token-Endpoint) of the [OAuth 2.0 Authorization Code Grant Flow](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_158074210415.html#OAuth-2.0-Authorization-Code-Grant-Flow) tu ...
+| accessTokenUrl | The [!DNL NetSuite] Access Token URL. This takes the form shown alongside. | <code>https://<NETSUITE_ACCOUNT_ID>.suitetalk.api.netsuite.com<br>/services/rest/auth/oauth2/v1/token</code> |
```
sentence case for mentions of "step two"

carlojavier commented on help/sources/connectors/marketing-automation/oracle-netsuite.md:
```diff
@@ -0,0 +1,73 @@
+
+>[!IMPORTANT]
+>
+> After a Refresh token expires you must create a New Account in Platform with updated tokens.
```
sentence case for "refresh token"

carlojavier commented on help/sources/tutorials/api/create/marketing-automation/oracle-netsuite-activities.md:
````diff
@@ -0,0 +1,646 @@
+
+A successful response returns a JSON structure like the following: 
+
+```json
````
Let's wrap this response in a collapsable (+++) and add "Select to view JSON payload" as a prompt

carlojavier commented on help/sources/tutorials/api/create/marketing-automation/oracle-netsuite-entities.md:
```diff
@@ -0,0 +1,995 @@
+
+>[!TAB Customer]
+
+A successful response returns a structure as below. 
```
Let's wrap these responses in a collapsable as well (+++), they should work within the tabs

carlojavier commented on help/sources/tutorials/api/create/marketing-automation/oracle-netsuite-entities.md:
````diff
@@ -0,0 +1,995 @@
+```
+
+>[!TAB Contact]
+
````
Let's add a collapsable here

carlojavier commented on help/sources/tutorials/ui/create/marketing-automation/oracle-netsuite-activities.md:
```diff
@@ -0,0 +1,98 @@
+| authorizationTestUrl | | https://<ACCOUNT_ID>.app.netsuite.com/app/login/oauth2/authorize.nl?response_type=code&redirect_uri=https%3A%2F%2Fapi.github.com&scope=rest_webservices&state=ykv2XLx1BpT5Q0F3MRPHb94j&client_id=<CLIENT_ID> |
+| Access token | The Access token value is generated at the end of [Step Two](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_158081952044.html#Step-Two-POST-Request-to-the-Token-Endpoint) of the [OAuth 2.0 Authorization Code Grant Flow](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_158074210415.html#OAuth-2.0-Authorization-Code-Grant-Flow) tutorial.  ...
+| Refresh token | The Refresh token value value is generated at the end of [Step Two](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_158081952044.html#Step-Two-POST-Request-to-the-Token-Endpoint) of the [OAuth 2.0 Authorization Code Grant Flow](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_158074210415.html#OAuth-2.0-Authorization-Code-Grant-Flow) tu ...
+| accessTokenUrl | The [!DNL NetSuite] Access Token URL. This takes the form shown alongside. | https://<ACCOUNT_ID>.suitetalk.api.netsuite.com/services/rest/auth/oauth2/v1/token |
```
Let's reformat the placeholder values in the examples from "<EXAMPLE>" to "{EXAMPLE}" please apply throughout the other docs too

carlojavier reviewed (changes requested):
@mohan-h thanks for the submission! I've left some comments for you

mohan-h committed:
Merge remote-tracking branch 'upstream/main' into oracle-netsuite-activities

mohan-h committed:
Corrections to address review comments

https://github.com/AdobeDocs/experience-platform.en/pull/281/#pullrequestreview-1775985925
